staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -367,7 +471,7 @@ func (cnc *CloudNodeController) initializeNode(ctx context.Context, node *v1.Nod
 	// get instanceMetadata and pass it to getNodeModifiersFromCloudProvider and updateNodeAddress which reduces api calls.
 	nodeModifiers, err := cnc.getNodeModifiersFromCloudProvider(ctx, curNode)
```

Comment:
We should deep copy the node object here before passing it into the node modifiers, since you shouldn't mutate cached objects
```
copyNode := curNode.DeepCopy()
nodeModifiers, err := cnc.getNodeModifiersFromCloudProvider(ctx, copyNode)
if err != nil {
  utilruntime.HandleError(fmt.Errorf("failed to initialize node %s at cloudprovider: %v", nodeName, err))
  return
}
```

Reply:
gotcha, will update!